internal/model/model.go

```diff
@@ -1129,19 +1132,33 @@ func (m *Model) ScanFolderSub(folder, sub string) error {
 
 	// Required to make sure that we start indexing at a directory we're already
 	// aware off.
-	for sub != "" {
-		if _, ok = fs.Get(protocol.LocalDeviceID, sub); ok {
-			break
+	var unifySubs []string
+	for _, sub := range subs {
+		for sub != "" {
+			if _, ok = fs.Get(protocol.LocalDeviceID, sub); ok {
+				break
+			}
+			sub = filepath.Dir(sub)
+			if sub == "." || sub == string(filepath.Separator) {
+				sub = ""
+			}
 		}
-		sub = filepath.Dir(sub)
-		if sub == "." || sub == string(filepath.Separator) {
-			sub = ""
+		seenPrefix := false
+		for _, us := range unifySubs {
+			if strings.HasPrefix(sub, us) {
+				seenPrefix = true
+				break
```

Comment:
Currently this sort&prefix check is already performed in the inotify client to aggregate changes. Before this patch it was important to have as few http calls as possible.
I don't know if repeating it here is useful, maybe it could be of use for other clients? However I think we don't need the optimisation here as we could introduce extra intermediate data for the GC to collect.

Note for the record: I don't like goto labels but you forced me:) It looks a bit cleaner but it's scary ^^